### PR TITLE
Alternate job titles

### PR DESCRIPTION
**Pull request by BootlegBow**
## About The Pull Request

Adds alternate job titles, plus senior job titles (>50 hours) for some jobs.

Port of https://github.com/Whitesands13/Whitesands/pull/251

## Why It's Good For The Game

Roleplay, and it adds a little more variety. Also just a generally nice thing to have.

## Changelog
:cl: MarkSuckerberg, BootlegBow
add: Alt job titles
add: Senior job titles
/:cl:

### CODE REVIEW

**Comment by BootlegBow** [comment]
alright i could swear there were no errors when i compiled it

**Comment by BootlegBow** [comment]
if anyone has any idea how to fix the undefined variable errors that'd be real nice cause i have no clue

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -519,9 +519,12 @@
 	if((character.mind.assigned_role == "Cyborg") || (character.mind.assigned_role == character.mind.special_role))
 		return
 
+	var/displayed_rank = rank//Tegu edit - Alt job titles
```
Comment (JohnFulpWillard) [inline code/__HELPERS/game.dm:522]: Should make the comment here clearer, perhaps put it at the beginning, like for the edit end comment

File: code/datums/datacore.dm [hunk 2]
```diff
@@ -296,7 +304,7 @@
 		var/datum/data/record/L = new()
 		L.fields["id"]			= md5("[H.real_name][H.mind.assigned_role]")	//surely this should just be id, like the others?
 		L.fields["name"]		= H.real_name
-		L.fields["rank"] 		= H.mind.assigned_role
```
Comment (JohnFulpWillard) [inline code/datums/datacore.dm:299]: Is this edit necessary?

File: code/modules/mob/dead/new_player/new_player.dm [hunk 3]
```diff
@@ -408,13 +408,16 @@
...
 			if(job_datum && IsJobUnavailable(job_datum.title, TRUE) == JOB_AVAILABLE)
+				var/altjobline = "" //tegu edit - alt job titles
 				var/command_bold = ""
+				if(client && client.prefs && client.prefs.alt_titles_preferences[job_datum.title])//tegu edit - alt job titles
+					altjobline = "(as [client.prefs.alt_titles_preferences[job_datum.title]])"//tegu edit - alt job titles
 				if(job in GLOB.command_positions)
 					command_bold = " command"
 				if(job_datum in SSjob.prioritized_jobs)
-					dept_dat += "<a class='job[command_bold]' href='byond://?src=[REF(src)];SelectedJob=[job_datum.title]'><span class='priority'>[job_datum.title] ([job_datum.current_positions])</span></a>"
+					dept_dat += "<a class='job[command_bold]' href='byond://?src=[REF(src)];SelectedJob=[job_datum.title]'><span class='priority'>[job_datum.title] [altjobline] ([job_datum.current_positions])</span></a>"
```
Comment (JohnFulpWillard) [inline code/modules/mob/dead/new_player/new_player.dm:418]: Missing comments on these lines

File: code/modules/jobs/job_types/_job.dm [hunk 4]
```diff
@@ -269,7 +274,7 @@
 		holder = "[uniform]"
 	uniform = text2path(holder)
 
-/datum/outfit/job/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+/datum/outfit/job/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE, client/preference_source = null)
```
Comment (JohnFulpWillard) [inline code/modules/jobs/job_types/_job.dm:277]: Missing comment here

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 5]
```diff
@@ -285,7 +285,7 @@
 	sec_hud_set_security_status()
 	..()
 
-/mob/living/carbon/human/proc/equipOutfit(outfit, visualsOnly = FALSE)
+/mob/living/carbon/human/proc/equipOutfit(outfit, visualsOnly = FALSE, client/preference_source = null)
```
Comment (JohnFulpWillard) [inline code/modules/mob/living/carbon/human/inventory.dm:288]: Missing comments here too

File: code/modules/client/preferences.dm [hunk 6]
```diff
@@ -1161,6 +1168,24 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 						joblessrole = BERANDOMJOB
 					if(BERANDOMJOB)
 						joblessrole = RETURNTOLOBBY
+				SetChoices(user)//tegu edit alt job titles
+			if("alt_title")
+				var/job_title = href_list["job_title"]
+				var/titles_list = list(job_title)
+				var/datum/job/J = SSjob.GetJob(job_title)
+				if(user.client.prefs.exp[job_title] > (J.get_exp_req_amount() + 3000)) //If they have more than 50 hours (300 Minutes) past the required time needed for the job, give them access to the senior title
```
Comment (JohnFulpWillard) [inline code/modules/client/preferences.dm:1176]: Is timelocking job titles (which give IC benefit) good in any way? Heads of staff makes sense because there's IC benefit behind timelocked jobs, but it comes with a lot of extra responsibility, so the timelock is to make sure you can handle it. Here it's just "here you go, free stuff"

File: ModularTegustation/altjobtitles/outfits.dm [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+	mask = /obj/item/clothing/mask/surgical
+
+/datum/outfit/job/doctor/nurse
+ 	name = "Medical Doctor (Nurse)"
+ 	head = /obj/item/clothing/head/nursehat
+ 	suit = null
+ 	accessory = /obj/item/clothing/accessory/armband/medblue
+
+/datum/outfit/job/doctor/psychiatrist
+	name = "Medical Doctor (Psychiatrist)"
```
Comment (JohnFulpWillard) [inline ModularTegustation/altjobtitles/outfits.dm:76]: isnt this just a psychologist?

File: ModularTegustation/altjobtitles/outfits.dm [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+		/obj/item/tape = 1
+	)
+
+//lawyer
+/datum/outfit/job/lawyer/corporaterepresentative
+	uniform = /obj/item/clothing/under/suit/black
+	suit = /obj/item/clothing/suit/toggle/lawyer/black
+	neck = /obj/item/clothing/neck/tie/blue
+	l_hand = /obj/item/clipboard
+	r_pocket = /obj/item/pen/fountain
```
Comment (JohnFulpWillard) [inline ModularTegustation/altjobtitles/outfits.dm:60]: Their PDA already has a fountain pen

File: ModularTegustation/altjobtitles/outfits.dm [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+	name = "Assistant (Businessman)"
+	uniform = /obj/item/clothing/under/suit/black_really
+	l_hand = /obj/item/storage/briefcase
+
+/datum/outfit/job/assistant/visitor
+	name = "Assistant (Visitor)"
+	uniform = /obj/item/clothing/under/misc/assistantformal
+	neck = /obj/item/camera
+
+/datum/outfit/job/assistant/trader
```
Comment (JohnFulpWillard) [inline ModularTegustation/altjobtitles/outfits.dm:12]: Maybe give them a Cargo tech tablet too

File: ModularTegustation/altjobtitles/altjobtitles.dm [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+	alt_titles = list("Jester", "Comedian")
+	senior_title = "Master Prankster"
+
+/datum/job/mime
+	alt_titles = list("Performer")
+	senior_title = "True Frenchman"
+
+ //Science
+/datum/job/rd
+	alt_titles = list("Research Supervisor", "Head Scientist", "Chief Researcher", "Chief Science Officer")
```
Comment (JohnFulpWillard) [inline ModularTegustation/altjobtitles/altjobtitles.dm:57]: "Chief Science Officer"
that killed me internally

File: ModularTegustation/altjobtitles/altjobtitles.dm [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+/datum/job/curator
+	alt_titles = list("Journalist", "Librarian")
+
+/datum/job/janitor
+	alt_titles = list("Custodian", "Sanitation Technician")
+
+/datum/job/lawyer
+	alt_titles = list("Corporate Representative", "Attorney")
+	senior_title = "Ace Attorney"
```
Comment (JohnFulpWillard) [inline ModularTegustation/altjobtitles/altjobtitles.dm:42]: oh god

**Review by JohnFulpWillard — APPROVED** [review]

**Comment by JohnFulpWillard** [comment]
FAIL: /datum/unit_test/outfit_sanity 6.4s
	REASON #1: Medical Doctor (Surgeon)'s suit_store is invalid!
	REASON #2: Medical Doctor (Nurse)'s suit_store is invalid!
	REASON #3: Medical Doctor (Psychiatrist)'s suit_store is invalid!

File: code/modules/client/preferences.dm (hunk 6 above)
Comment (BootlegBow) [reply code/modules/client/preferences.dm:1176]: From what I can tell, it was originally to differentiate between new-er players and ones that have been around a while

**Review by EgorDinamit — APPROVED** [review]
Merging this in a few days if nobody has other concerns or @BootlegBow spots issues with it while testing on a local server.

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
I'd ask you to remove "Junior [Job]" titles, these feel obsolete.
I've tested this PR on the local server and it seems that HUDs work as expected, but late-joins **AND** round-start spawns won't assign your alternative title, although, it will give you the loadout for the selected title(businessman, for instance).
I'd also appreciate it if the amount of time for the senior title to unlock was a config value.

**Comment by BootlegBow** [comment]
> I'd ask you to remove "Junior [Job]" titles, these feel obsolete.
I thought you already did that in a previous commit

**Comment by EgorDinamit** [comment]
> > I'd ask you to remove "Junior [Job]" titles, these feel obsolete.
> I thought you already did that in a previous commit

Some of them, yes, but I missed some.
Anyway, first things first - it has to work properly.

**Review by EgorDinamit — APPROVED** [review]
This should be okay now.

File: ModularTegustation/altjobtitles/altjobtitles.dm [hunk 12]
```diff
@@ -14,10 +20,10 @@
 	alt_titles = list("Firefighter", "Life Support Specialist")
 	senior_title = "Senior Atmospheric Technician"
 
- //Service
+ // Service
 /datum/job/assistant
 	alt_titles = list("Visitor", "Businessman", "Entertainer")
-	senior_title = "Greytider"
+	ultra_senior_title = "Assistinator"
```
Comment (BootlegBow) [inline ModularTegustation/altjobtitles/altjobtitles.dm:26]: oh my god egor

Comment (EgorDinamit) [reply ModularTegustation/altjobtitles/altjobtitles.dm:26]: Yes. People(including one admin) were in favor of this.

Comment (BootlegBow) [reply ModularTegustation/altjobtitles/altjobtitles.dm:26]: image
i love this community

File: config/config.txt [hunk 13]
```diff
@@ -541,3 +541,9 @@ DEFAULT_VIEW_SQUARE 15x15
 
 ## Add the ID of the role you want assigning here
 #DISCORD_ROLEID 000000000000000000
+
+##Amount of time required for senior job titles in minutes
+SENIOR_TIMELOCK 3000
+
+##Amount of time required for ULTRA senior job titles in minutes
+SENIOR_TIMELOCK 30000
```
Comment (JohnFulpWillard) [inline config/config.txt:549]: i really dislike gatekeeping. 500 hours in a single job? My most played job is 300

Comment (EgorDinamit) [reply config/config.txt:549]: It's a config option for a reason.
